### PR TITLE
Add blog module to olympia

### PR DESCRIPTION
**Pull request by conectado**
# This PR targets #1860

These are the things it adds:
* Add blog module into olympia
* Update substrate version for blog module
* Make blog module isntanceable and remove dynamic instances
* Update blog module ownership to be used with proposals instead
* Store hashes instead of text
* Update reactions to work as in forum module
* Adds the new proposals
* Benchmarks the blog module
* Discard blog locking(Post locking is still working through proposals)
* Add blog module into the runtime
* Update blog module to use membership for allowed participants
* Updates tests according to these changes

Corresponding companion [handbook PR](https://github.com/Joystream/handbook/pull/30)

### CODE REVIEW

**Comment by shamil-gadelshin** [comment]
Are you able to run tests with the "all-features" flag? @conectado 

I got this error:
```
error[E0407]: method `add` is not a member of trait `$crate::traits::Contains`
   --> runtime-modules/blog/src/mock.rs:215:1
    |
215 | / ord_parameter_types! {
216 | |     pub const CorrectOwner: u64 = FIRST_OWNER_ORIGIN;
217 | | }
    | |_^ not a member of trait `$crate::traits::Contains`
```

**Comment by conectado** [comment]
> Are you able to run tests with the "all-features" flag? @conectado
> 
> I got this error:
> 
> ```
> error[E0407]: method `add` is not a member of trait `$crate::traits::Contains`
>    --> runtime-modules/blog/src/mock.rs:215:1
>     |
> 215 | / ord_parameter_types! {
> 216 | |     pub const CorrectOwner: u64 = FIRST_OWNER_ORIGIN;
> 217 | | }
>     | |_^ not a member of trait `$crate::traits::Contains`
> ```

Only tested with `--features runtime-benchmarks` I'll try with the flag `all-features` now.

File: runtime-modules/blog/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+[package]
+name = "pallet-blog"
+version = "3.0.0"
+authors = ["iorveth"]
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/Cargo.toml]: Incorrect authors: should be 'Joystream contributors'

File: runtime-modules/blog/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        assert_eq!(Blog::<T, I>::post_count(), 0);
+        let title = vec![0u8; t.try_into().unwrap()];
+        let body = vec![0u8; b.try_into().unwrap()];
+
+    }:_(RawOrigin::Root, title.clone(), body.clone())
+    verify {
+        assert_eq!(Blog::<T, I>::post_count(), 1);
+
+        assert!(
+            Blog::<T, I>::post_by_id(0) ==
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/src/benchmarking.rs]: Please, introduce a post_id variable here.

File: runtime-modules/blog/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+            assert_ok!(test_benchmark_edit_reply::<Runtime>());
+        })
+    }
+
+    #[test]
+    fn test_react_to_post() {
+        ExtBuilder::default().build().execute_with(|| {
+            assert_ok!(test_benchmark_react_to_post::<Runtime>());
+        })
+    }
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/src/benchmarking.rs]: It seems, that `react_to_reply` test is missing.

File: runtime-modules/blog/src/benchmarking.rs [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+    edit_post {
+        let t in 0 .. MAX_BYTES;
+        let b in 0 .. MAX_BYTES;
+
+        let post_id = generate_post::<T, I>();
+        let title = Some(vec![1u8; t.try_into().unwrap()]);
+        let body = Some(vec![1u8; b.try_into().unwrap()]);
+    }: _(
+        RawOrigin::Root,
```
Comment (shamil-gadelshin) [inline runtime-modules/blog/src/benchmarking.rs]: Do we need such formatting in this and other blog benchmarks? It looks different from the `cargo fmt` style.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Benchmarking looks very good!

Several cosmetic comments:
- please add `weights` comments to all extrinsics.
- use `assert_eq!` macro instead of `assert!` with ==
- format extrinsic calls (and other) according to `cargo fmt` style (with 99 chars limit in mind)

File: runtime-modules/blog/src/benchmarking.rs (hunk 3 above)
Comment (conectado) [reply runtime-modules/blog/src/benchmarking.rs]: Good catch!

**Review by bedeho — CHANGES_REQUESTED** [review]
Excellent work, very clean.

I think in retrospect we should just drop everything having to do with reactions, I do not believe it will be used and if it is it just adds perpetual state of relatively little value.

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Review by bedeho — APPROVED** [review]
Great job.

**Pull request merged**